### PR TITLE
update bioconductor packages to bioconductor 3.14

### PR DESCRIPTION
**Pull request by glennpj**
Depends on #28786

This PR does a sweep of Bioconductor R packages in Spack, and does the
following:

- add new versions of packages for bioconductor 3.14
- make sure the bioc attribute is set
- add new packages as required

Due to adding the bioc attribute, almost all R bioconductor packages had
changes. Since almost all packages were being touched, I also tried to
standardize the formatting of the packages.

- make sure there is a short and long description
- have dependencies specified in the same order as in the DESCRIPTION
  file
- place when directives at the end

Having consistency makes it easier to process these in bulk. The process
is script driven but requires interaction. I am hopeful that the
consistent formatting will allow for further automation. All of the
packages in this PR were built with gcc-10.3.0 prior to submitting the
PR.

### CODE REVIEW

File: var/spack/repos/builtin/packages/r-do-db/package.py [hunk 1]
```diff
@@ -7,15 +7,16 @@
 
 
 class RDoDb(RPackage):
-    """A set of annotation maps describing the entire Disease Ontology
+    """A set of annotation maps describing the entire Disease Ontology.
 
     A set of annotation maps describing the entire Disease
     Ontology assembled using data from DO."""
 
+    # There is no git repoistory for this package.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-do-db/package.py]:
```suggestion
    # There is no git repository for this package.
```

File: var/spack/repos/builtin/packages/r-genefilter/package.py [hunk 2]
```diff
@@ -7,13 +7,13 @@
 
 
 class RGenefilter(RPackage):
-    """genefilter: methods for filtering genes from high-throughput experiments
+    """: methods for filtering genes from high-throughput experiments.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-genefilter/package.py]: I would undo this removal of "genefilter"

File: var/spack/repos/builtin/packages/r-ggtree/package.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    """an R package for visualization of tree and annotation data.
+
+    'ggtree' extends the 'ggplot2' plotting system which implemented the
+    grammar of graphics. 'ggtree' is designed for visualization and annotation
+    of phylogenetic trees and other tree-like structures with their annotation
+    data."""
+
+    bioc = "ggtree"
+
+    version('3.2.1', commit='d3747e6')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-ggtree/package.py]: Could we use a full commit hash here?

File: var/spack/repos/builtin/packages/r-treeio/package.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    'treeio' is an R package to make it easier to import and store phylogenetic
+    tree with associated data; and to link external data from different sources
+    to phylogeny. It also supports exporting phylogenetic tree with
+    heterogeneous associated data to a single tree file and can be served as a
+    platform for merging tree with associated data and converting file
+    formats."""
+
+    bioc = "treeio"
+
+    version('1.18.1', commit='a06b6b3')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-treeio/package.py]: Can we use a full commit hash here?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/r-ggtree/package.py (hunk 3 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/r-ggtree/package.py]: Sure, there might be a few of those.

File: var/spack/repos/builtin/packages/r-genefilter/package.py (hunk 2 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/r-genefilter/package.py]: Yep, I meant to remove those but missed the colon for this one.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**